### PR TITLE
Fix lots of small issues reported by staticcheck

### PR DESCRIPTION
**Pull request by MichaelEischer**
What is the purpose of this change? What does it change?
--------------------------------------------------------

Cleanup a few code style issues found by staticcheck.

There are a few noteworthy changes though:
- `archiver_test: Only skip symlinks on windows` The test contained one `continue` too much and skip too much on windows
- `restic/file: Use FileType for all File constants`

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------
No.

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- ~~[ ] I have added tests for all changes in this PR~~
- ~~[ ] I have added documentation for the changes (in the manual)~~
- No user facing changes ~~[ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))~~
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: internal/backend/rclone/backend.go [hunk 1]
```diff
@@ -212,12 +212,11 @@ func New(cfg Config, lim limiter.Limiter) (*Backend, error) {
 	// rclone is able to accept HTTP requests.
 	url := fmt.Sprintf("http://localhost/file-%d", rand.Uint64())
 
-	req, err := http.NewRequest(http.MethodGet, url, nil)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
```
Comment (greatroar) [inline internal/backend/rclone/backend.go]: This was added in Go 1.13.

File: internal/repository/worker_group.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ import (
 // one of the workers, it is returned. FinalFunc is always run, regardless of
 // any other previous errors.
 func RunWorkers(ctx context.Context, count int, workerFunc func() error, finalFunc func()) error {
-	wg, ctx := errgroup.WithContext(ctx)
+	wg, _ := errgroup.WithContext(ctx)
```
Comment (greatroar) [inline internal/repository/worker_group.go]: If the context is unused, this is equivalent to `var wg errgroup.Group`.

File: internal/backend/rclone/backend.go (hunk 1 above)
Comment (MichaelEischer) [reply internal/backend/rclone/backend.go]: I've dropped the corresponding commit.

File: internal/repository/worker_group.go (hunk 2 above)
Comment (MichaelEischer) [reply internal/repository/worker_group.go]: Thanks, good catch.

**Review by greatroar — APPROVED** [review]
Confirmed all changes except the last one, but if the CI doesn't error on the expanded Windows test it should be fine.

**Comment by MichaelEischer** [comment]
Rebased to fix the conflict with the random reader cleanup.

@greatroar I didn't test the last change, but since the test seems to run successfully on AppVeyor is assume it to be fine. And otherwise this would be a bug that needed fixing as the previous behavior of skipping everything was clearly not intended.

**Review by fd0 — APPROVED** [review]
LGTM! Thanks a lot for splitting all of this into small commits, which makes it a lot easier to review!

**Pull request merged**